addon/utils/computed.js

```diff
@@ -30,6 +31,10 @@ export function guidFor(dependentKey) {
     get() {
       const value = get(this, dependentKey);
 
+      // it's possible that a value has no toString as some objects don't implement the guid field
+      // this early return it to avoid errors being thrown when calling undefined.toString()
+      if (isNone(value)) return;
```

Comment:
Perhaps just checking `if (typeof value.toString !== "function")` would show your intention more obviously?

Reply:
Sadly it would require additional checks as `typeof undefined.toString` will throw an error:

```
Uncaught TypeError: Cannot read property 'toString' of undefine
```